base/cvd/cuttlefish/host/libs/metrics/session_id.cc

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cuttlefish/host/libs/metrics/session_id.h"
+
+#include <string>
+#include <string_view>
+
+#include <uuid/uuid.h>
+
+#include "cuttlefish/common/libs/utils/files.h"
+#include "cuttlefish/common/libs/utils/result.h"
+
+namespace cuttlefish {
+namespace {
+
+constexpr char kSessionIdFileName[] = "metrics_session_id.txt";
+
+std::string GenerateUuid() {
+  uuid_t uuid;
+  uuid_generate_random(uuid);
+  std::string uuid_str = "xxxxxxxx-xxxx-Mxxx-Nxxx-xxxxxxxxxxxx";
```

Comment:
According to `UUID_UNPARSE(3)` the only requirement on this string is that it is 36 chars long plus the `\0` at the end. The pattern you've included here guarantees that, but it's error prone as someone could inadvertently delete part of it and not notice it. Consider building the string with something like `std::string(36, 'x')`, even better if you make that 36 a constant.